source/include-link.txt

```diff
@@ -0,0 +1,209 @@
+
+.. code-block:: none
+
+   $ cmake -Bbuild -DCMAKE_CXX_STANDARD=17 -DCMAKE_PREFIX_PATH=/opt/mongodb/cxx-driver
+
+Without Driver Installation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Alternatively, you can use CMake's ``add_subdirectory`` command without installing the {+driver-short+}. This is an advanced technique that, unlike the  ``find_package`` command, does not support specifying version constraints. There are additional drawbacks associated with ``add_subdirectory``, but discussion of those is beyond the scope of this document.
+
+.. note::
+
+   You may need to adjust build flags for your project to
+   accommodate this approach. Your project's invocation of CMake must include any flags or
+   options that you would have passed to the {+driver-short+} as part of its build.
+
+To use the ``add_subdirectory`` command, create a ``CMakeLists.txt`` file in your project directory. The following example creates a ``CMakeLists.txt`` file in the ``/home/user/project2`` project directory that uses ``add_subdirectory``:
+
+.. literalinclude:: /includes/cmake_without_driver_installation.txt
+  :caption: /home/user/project2/CMakeLists.txt
+  :start-after: -- sphinx-include-start --
+
+.. note::
+
+   Note that in this approach the CMake target is referred to simply
+   as ``bsoncxx_shared``, without the ``mongo::`` namespace. The namespace
+   is added as part of the CMake module installation, which is not performed
+   in this approach.
+
+Then, you can build and run your project. The following shows a truncated example of the CMake output and the program execution output immediately following:
+
+.. code-block:: none
+
+   $ cmake -Bbuild
+   $ cmake --build build
+   $ ./build/view_and_value 
+   { "team" : "platforms", "id" : { "$oid" : "67207dcf532837a4470cc090" }, "members" : [ "tyler", "jason", "drew", "sam", "ernie", "john", "mark", "crystal" ] }
+   Got key, key = team
+   Got String!
+   Got key, key = id
+   Got ObjectId!
+   Got key, key = members
+   Got Array!
+   array element: tyler
+   array element: jason
+   array element: drew
+   array element: sam
+   array element: ernie
+   array element: john
+   array element: mark
+   array element: crystal
+   as expected, we have a team
+   document has 3 keys.
+   document keys are: 
+   team 
+   id 
+   members
+
+If you encounter errors, then refer to the preceding section, which includes suggested additional CMake options for resolving common build errors.
+
+pkg-config
+----------
+
+If your project is not CMake-based, you can use ``pkg-config`` to integrate the {+driver-short+} with your project. Because pkg-config provides less flexibility than CMake, we recommend preferring the CMake-based approach when possible.
```

Comment:
S: I think "using" is a slightly better word here than "preferring"
```suggestion
If your project is not CMake-based, you can use ``pkg-config`` to integrate the {+driver-short+} with your project. Because pkg-config provides less flexibility than CMake, we recommend using the CMake-based approach when possible.
```